verif/sim/cva6.py

```diff
@@ -954,6 +961,27 @@ def load_config(args, cwd):
 
   args.spike_params = get_full_spike_param_args(args.spike_params) if args.spike_params else ""
 
+def edit_Makefile(UVM_TESTNAME="uvmt_cva6_firmware_test_c"):
```

Comment:
I think it would be better to pass the uvm test to the Makefile as an env var instead of rewriting it at each simulation. This is how it is done for such things like the test path.
This is how you can do this : 

- add a `uvm_test `parameter to the `run_test` function in `cva6.py` (I admit that this function has too much parameters...)
- also add it in `get_iss_cmd` function 
- substitute it in the cmd like it is done for the elf path for instance
- add a `uvm_test `env var for the uvm targets in `verif/sim/cva6.yaml`
- set a default value in `verif/sim/Makefile` for the environment var `uvm_test`
- replace `UVM_TESTNAME=uvmt_qqchose` with `UVM_TESTNAME=$(uvm_test)` in `verif/sim/Makefile`

Do not hesitate to ask for more details if needed

Reply:
I guess we could call each others if you think this could speed up things

Reply:
@OlivierBetschi I wonder why you use a different uvm test than "uvmt_cva6_firmware_test_c" ?
@AnouarZajni, can you help us to understand what is the best solution ?

Reply:
The different uvm_test came from the need to disable the cache for the PMP testing (that is my understanding at least).